webgui/src/flipmouse/js/sercomm.js

```diff
@@ -15,141 +15,129 @@ function SerialCommunicator() {
     this.setValueHandler = function (handler) {
         _valueHandler = handler;
     };
-    
-    function ping(port_name){
-		try {
-		   const port = new SerialPort(port_name,{
-			 baudRate: 115200,
-			 parser: new SerialPort.parsers.Readline('\n')
-		   });
-		   port.on('open',function(){
-		     port.write("AT ID \r\n");
-		     data = port.read();
-		     //TODO: test for valid id
-		     console.log(i + data);
-		     port.close();
-		  });
-		
-		  return new Promise(resolve => port.on("close", resolve));
-	   } catch(err) {
-		   console.log("Not a valid port");
-		   return new Promise(function (resolve, reject) { 
-			   reject();
-		   });
-	   }
-	
-	   
-	}
-	
-    
-    this.connect = async function() {
-		 
 
-			let portname = await this.getValidPort();
-			
-			serialport = "/dev/ttyACM0";
-			
-			
-		 //})();
-		
-		/*(async function() {
-			const portlist = await SerialPort.list();
-			let serialport = "";
-			for(const port of portlist) {
-				//open port
-				probePort = new SerialPort(port['comName'], {baudRate: 115200 })
-				probePort.pipe(parser);
-				
-				//send data
-				
-				//await  for feedback
-				
-				// if flipmouse -> set _port & return
-			}*/
-			
-			
-			
-			//is port defined?
-			
-			//attach parser to serial port
-			_port = new SerialPort(serialport, {
-				baudRate: 115200 })
-			_port.pipe(parser);
-			console.log("Found valid device @" + serialport);
-				    
-        
-			//on a fully received line,
-			//check if the data is a reported raw value
-			//or returned data for a dedicated command (see sendData)
-			parser.on('data', function(data) {
-				console.log("data evt: " + data);
+    function isATCOM({ comName }, device = 'FABI v2.3') {
+        return new Promise((resolve, reject) => {
+            const port = new SerialPort(
+                comName,
+                {
+                    baudRate: 115200,
+                    parser: new SerialPort.parsers.Readline('\n')
+                },
+                function(error) {
+                    if (!error) {
+                        let found = false;
+                        port.on('data', chunk => {
+                            const msg = chunk.toString();
+                            if (msg.startsWith(device)) {
+                                found = true;
+                                console.log(
+                                    `Found AT COM device at ${comName}`
+                                );
+                                port.close();
```

Comment:
Maybe we use the parser event for a fully received line?
parser.on('data' ...

Reply:
Oh. I didn't know about that possibility. I think you're right. It's probably better to safeguard the communication/transmission somehow.